### PR TITLE
Mon 34072 exceptions

### PR DESCRIPTION
**Pull request by bouda1**
## Description

The PR follows MON-34072-dependencies-escalations-reviewed

REFS: MON-34072

### CODE REVIEW

File: engine/inc/com/centreon/engine/configuration/anomalydetection.hh [hunk 1]
```diff
@@ -31,6 +30,18 @@ namespace configuration {
 
 class anomalydetection : public object {
  public:
+  enum tag_type {
```
Comment (jean-christophe81) [inline engine/inc/com/centreon/engine/configuration/anomalydetection.hh:33]: yet defined in tag.proto

File: engine/inc/com/centreon/engine/configuration/anomalydetection.hh [hunk 2]
```diff
@@ -31,6 +30,18 @@ namespace configuration {
...
+    SERVICEGROUP = 0,
+    HOSTGROUP = 1,
+    SERVICECATEGORY = 2,
+    HOSTCATEGORY = 3,
+  };
+  enum service_state {
+    OK = 0,
```
Comment (jean-christophe81) [inline engine/inc/com/centreon/engine/configuration/anomalydetection.hh:40]: yet defined in neb.proto

File: engine/inc/com/centreon/engine/configuration/host.hh [hunk 3]
```diff
@@ -33,6 +32,17 @@ namespace configuration {
 
 class host : public object {
  public:
+  enum tag_type {
+    SERVICEGROUP = 0,
```
Comment (jean-christophe81) [inline engine/inc/com/centreon/engine/configuration/host.hh:36]: yet defeined in tag.proto

File: engine/inc/com/centreon/engine/configuration/host.hh [hunk 4]
```diff
@@ -33,6 +32,17 @@ namespace configuration {
...
+    HOSTGROUP = 1,
+    SERVICECATEGORY = 2,
+    HOSTCATEGORY = 3,
+  };
+  enum host_state {
```
Comment (jean-christophe81) [inline engine/inc/com/centreon/engine/configuration/host.hh:41]: yet defined in neb.proto

File: engine/src/configuration/hostgroup.cc [hunk 5]
```diff
@@ -193,8 +188,8 @@ hostgroup::key_type const& hostgroup::key() const throw() {
  */
 void hostgroup::merge(object const& obj) {
   if (obj.type() != _type)
-    throw(engine_error() << "Cannot merge host group with '" << obj.type()
-                         << "'");
+    throw msg_fmt("Cannot merge host group with '",
```
Comment (jean-christophe81) [inline engine/src/configuration/hostgroup.cc:191]: {} missing

**Review by jean-christophe81 — APPROVED** [review]

**Comment by bouda1** [comment]
REFS: MON-34072

**Pull request merged**